opensrp-tasking/src/main/java/org/smartregister/tasking/presenter/TaskingHomePresenter.java

```diff
@@ -0,0 +1,825 @@
+            }
+        }
+    }
+
+    @Override
+    public void onEventFound(Event event) {
+        startForm(selectedFeature, cardDetails, selectedFeatureInterventionType, event);
+    }
+
+    @Override
+    public void findLastEvent(String featureId, String eventType) {
+    }
+
+    @Override
+    public void onFociBoundaryLongClicked() {
+    }
+
+    @Override
+    public void onInterventionFormDetailsFetched(CardDetails cardDetails) {
+        this.cardDetails = cardDetails;
+        this.changeInterventionStatus = true;
+        if (getView() != null) {
+            getView().hideProgressDialog();
+        }
+        if (validateFarStructures()) {
+            validateUserLocation();
+        } else {
+            onLocationValidated();
+        }
+    }
+
+    @Override
+    public TaskingHomeActivityContract.View getView() {
+        if (viewWeakReference != null) {
+            return viewWeakReference.get();
+        }
+        return null;
+    }
+
+    @Override
+    public void onInterventionTaskInfoReset(boolean success) {
+        if (getView() != null) {
+            getView().hideProgressDialog();
+            getView().closeAllCardViews();
+        }
+        if (taskingLibraryConfiguration.isRefreshMapOnEventSaved()) {
+            refreshStructures(true);
+            if (getView() != null) {
+                getView().clearSelectedFeature();
+            }
+            taskingLibraryConfiguration.setRefreshMapOnEventSaved(false);
+        }
+    }
+
+
+    @Override
+    public void onCardDetailsFetched(CardDetails cardDetails) {
+        if (getView() != null) {
+            getView().openCardView(cardDetails);
+        }
+    }
+
+    @Override
+    public void startForm(Feature feature, CardDetails cardDetails, String interventionType) {
+        startForm(feature, cardDetails, interventionType, null);
+    }
+
+    @Override
+    public void startForm(Feature feature, CardDetails cardDetails, String interventionType, Event event) {
+        String formName = jsonFormUtils.getFormName(null, interventionType);
+        String sprayStatus = cardDetails == null ? null : cardDetails.getStatus();
+        String familyHead = null;
+//        if (cardDetails instanceof SprayCardDetails) {
+//            familyHead = ((SprayCardDetails) cardDetails).getFamilyHead();
+//        }
+        startForm(formName, feature, sprayStatus, familyHead, event);
+    }
+
+    @Override
+    public void startForm(String formName, Feature feature, String sprayStatus, String familyHead, Event event) {
+        if (getView() != null) {
+            JSONObject formJson = jsonFormUtils.getFormJSON(getView().getContext()
+                    , formName, feature, sprayStatus, familyHead);
+            Map<String, Object> serverConfigsObjectMap = taskingLibraryConfiguration.getServerConfigs();
+//            if (cardDetails instanceof MosquitoHarvestCardDetails && PAOT.equals(((MosquitoHarvestCardDetails) cardDetails).getInterventionType())) {
```

Comment:
I am not sure of the intention when commenting these out. Could you do something about it? Create an issue, todo, remove, fix